palm/plugins/dbt/commands/cmd_test.py

```diff
@@ -1,42 +1,99 @@
 import click
 from typing import Optional, Tuple
 from palm.plugins.dbt.dbt_palm_utils import dbt_env_vars
+import sys
 
 
 @click.command('test')
 @click.option(
-    "--clean", is_flag=True, help="drop the test schema after the run is complete"
+    "--clean", is_flag=True, help="Drop the test schema after the run is complete"
 )
-@click.option("--models", multiple=True, help="see dbt docs on models flag")
-@click.option("--select", multiple=True, help="see dbt docs on select flag")
-@click.option("--exclude", multiple=True, help="see dbt docs on exclude flag")
-@click.option("--no-fail-fast", is_flag=True, help="will run all tests if one fails")
-@click.pass_context
+@click.option("--models", "-m", multiple=True, help="See dbt docs on models flag")
+@click.option("--select", "-s", multiple=True, help="See dbt docs on select flag")
+@click.option("--exclude", "-e", multiple=True, help="See dbt docs on exclude flag")
+@click.option("--defer", "-d", is_flag=True, help="See dbt docs on defer flag")
+@click.option(
+    "--no-fail-fast", "-nx", is_flag=True, help="Runs all tests even if one fails"
+)
+@click.pass_obj
 def cli(
-    ctx,
+    environment,
     clean: bool,
     no_fail_fast: bool,
+    defer: bool,
     models: Optional[Tuple] = tuple(),
     select: Optional[Tuple] = tuple(),
     exclude: Optional[Tuple] = tuple(),
 ):
-    """Tests the DBT repo"""
+    """Tests the dbt repo"""
+
+    if defer:
+        click.secho("Running 'palm prod-artifacts'...", fg='yellow')
+        exit_code, _, _ = environment.run_on_host("palm prod-artifacts")
+        if exit_code == 2:
+            click.secho(
+                "'palm prod-artifacts' not implemented. Can't use --defer without it!",
+                fg='red',
+            )
+            sys.exit(1)
+        elif exit_code != 0:
+            click.secho(
+                "Something went wrong while pulling the prod artifacts.", fg='red'
+            )
+            sys.exit(1)
+
+    env_vars = set_env_vars(environment, defer)
 
-    cmd = ['dbt', 'test']
-    if select:
+    # --select and --models are interchangeable on dbt >= v1, combine the lists of selections
+    targets = list(set(models + select))
+
+    run_cmd = build_run_command(
+        no_fail_fast=no_fail_fast,
+        targets=targets,
+        exclude=exclude,
+        defer=defer,
+    )
+    success, msg = environment.run_in_docker(run_cmd, env_vars)
+    click.secho(msg, fg="green" if success else "red")
+
+    if clean:
+        success, msg = environment.run_in_docker(
+            "dbt run-operation drop_branch_schemas", env_vars
+        )
+        click.secho(msg, fg="green" if success else "red")
+
+
+def build_run_command(
```

Comment:
```suggestion
def build_test_command(
```
There's also a possible abstraction here to `def build_command(base_command: str, ...` where `base_command` is either `run` or `test`. I don't think it adds a ton of value though, would be fine with just updating the name per the suggestion